Revit_Core_Engine/Query/DoesIntersect.cs

```diff
@@ -0,0 +1,176 @@
+                    if (intersectFilter.PassesFilter(element1))
+                        return true;
+                }
+            }
+            //link vs host
+            else if (doc1.IsLinked && !doc2.IsLinked)
+            {
+                BoundingBoxXYZ bbox2 = element2.PhysicalBounds();
+                Transform transform1 = doc1.LinkTransform();
+                BoundingBoxXYZ bbox1 = element1.get_Geometry(new Options()).GetTransformed(transform1).GetBoundingBox();
+
+                if (!bbox1.IsInRange(bbox2))
+                    return false;
+
+                List<Solid> element1Solids = element1.Solids(new Options());
+                foreach (Solid elSolid in element1Solids)
+                {
+                    Solid transformedSolid = SolidUtils.CreateTransformed(elSolid, transform1);
+                    ElementIntersectsSolidFilter intersectFilter = new ElementIntersectsSolidFilter(transformedSolid);
+                    if (intersectFilter.PassesFilter(element1))
+                        return true;
+                }
+            }
+            //link vs link
+            else if (doc1.IsLinked && doc2.IsLinked)
+            {
+                Transform transform1 = doc1.LinkTransform();
+                Transform transform2 = doc2.LinkTransform();
+                BoundingBoxXYZ bbox1 = element1.get_Geometry(new Options()).GetTransformed(transform1).GetBoundingBox();
+                BoundingBoxXYZ bbox2 = element2.get_Geometry(new Options()).GetTransformed(transform2).GetBoundingBox();
+
+                if (!bbox1.IsInRange(bbox2))
+                    return false;
+
+                List<Solid> element1Solids = element1.Solids(new Options());
+                List<Solid> element2Solids = element2.Solids(new Options());
+                List<Solid> el1TransSolids = element1Solids.Select(x => SolidUtils.CreateTransformed(x, transform1)).ToList();
+                List<Solid> el2TransSolids = element2Solids.Select(x => SolidUtils.CreateTransformed(x, transform2)).ToList();
+
+                foreach (Solid el1Solid in el1TransSolids)
+                {
+                    foreach (Solid el2Solid in el2TransSolids)
+                    {
+                        Solid intersection = BooleanOperationsUtils.ExecuteBooleanOperation(el1Solid, el2Solid, BooleanOperationsType.Intersect);
+                        if (intersection.Volume != 0)
+                            return true;
+                    }
+                }
+            }
+
+            return false;
+        }
+
+        /***************************************************/
+
+        [Description("Check if bounding box intersects with element.")]
+        [Input("bbox", "Bounding box to check the intersection for.")]
+        [Input("element", "Element to check the intersection for.")]
+        [Output("bool", "Result of the intersect checking.")]
+        public static bool DoesIntersect(this BoundingBoxXYZ bbox, Element element)
```

Comment:
This method reminds me of my first comment about the risk of bounding box transform being forgotten 😃